src/openrct2/entity/Peep.cpp

```diff
@@ -428,18 +428,12 @@ std::optional<CoordsXY> Peep::UpdateAction(int16_t& xy_distance)
         return UpdateWalkingAction(differenceLoc, xy_distance);
     }
 
-    const PeepAnimation& peepAnimation = GetPeepAnimation(SpriteType, ActionSpriteType);
-    ActionFrame++;
-
-    // If last frame of action
-    if (ActionFrame >= peepAnimation.frame_offsets.size())
+    if (!UpdateActionAnimation())
     {
         ActionSpriteImageOffset = 0;
         Action = PeepActionType::Walking;
         UpdateCurrentActionSpriteType();
-        return { { x, y } };
```

Comment:
Removing this alters the flow. It previously returned when at the last frame of action, but now it continues.

Reply:
If I'm not mistaken, that flow differences moved partially into `UpdateActionAnimation()`

Reply:
Decided to just reintroduce the return.